src/target/riscv/riscv.c

```diff
@@ -1636,59 +2070,56 @@ static int riscv_hit_watchpoint(struct target *target, struct watchpoint **hit_w
 	}
 
 	riscv_reg_t dpc;
-	riscv_get_register(target, &dpc, GDB_REGNO_DPC);
+	if (riscv_get_register(target, &dpc, GDB_REGNO_DPC) != ERROR_OK)
```

Comment:
I've looked at this MR in a much greater detail. I have doubts regarding this. I think that this early exit is incorrect:

```
for (struct watchpoint *wp = target->watchpoints; wp; wp = wp->next) {
		if (wp->unique_id == r->trigger_hit) {
			*hit_watchpoint = wp;
			return ERROR_OK;
		}
	}
```

It looks like that this condition should be reversed. If the HW does not support hit bit. we should not decode anything, since we don't have guarantee that we hit a watch point.

I think we should perform decode only if hit bit is supported.

Reply:
nevermind. I was wrong here. It seems that we just find the watchpoint, not the exact address. For the purposes of finding watch point - this is fine.